### ISSUE TITLE
package logpolicy doesn't seem to respect tsnet.Server.Dir

### ISSUE THREAD

**Issue by 08d2** [tailscale/tailscale#6605]
### What is the issue?

I have a Go program with

```go
ts := &tsnet.Server{
	Dir: "/var/myorg/my-tailscale-program",
	...
}
if err := ts.Start(); err != nil {
	return fmt.Errorf("start tailscale server: %w", err)
}
```

I run that program with a systemd .service file that looks like

```systemd
[Service]
User=mything
Group=mything
ExecStart=/opt/myorg/bin/my-tailscale-program
```

The host's /var/myorg/my-tailscale-program looks like

```shell
$ ls -ltar /var/myorg/my-tailscale-program
drwxr-xr-x 5 root      root      4096 Dec  1 21:26 ../
drwxr-xr-x 2 mything mything 4096 Dec  3 00:15 ./
```

But when I run the program, it says

```
start tailscale server:
 tsnet: 
  logpolicy.Config.Save for /tailscaled.log.conf: 
   open /tailscaled.log.conf.tmp2872976862: permission denied
```

It looks like it's trying to write to `/tailscaled.log.conf` instead of `/var/myorg/my-tailscale-program/tailscaled.log.conf`.

What did I do wrong?

### OS

Linux

### OS version

Debian 11

### Tailscale version

tailscale.com v1.32.3

### Bug report

BUG-895a74d7a948486d69cf0da0473372e0162fe69dd2d1715dc9983d3edbb2de9c-20221203002813Z-4a45a79e40c56e43

**Comment by DentonGentry** [comment]
The name of the logpolicy config file is set here: https://github.com/tailscale/tailscale/blob/07eacdfe923a2bd52b9f6f6a9a30d429638a083e/logpolicy/logpolicy.go#L493

Is the binary named `tailscaled`?

--------

The other place which might conceivably be trying to access /tailscaled.log.conf is some migration code handling very old Windows versions: https://github.com/tailscale/tailscale/blob/07eacdfe923a2bd52b9f6f6a9a30d429638a083e/logpolicy/logpolicy.go#L514

I don't see how that code could end up running on Debian.

**Comment by 08d2** [comment]
> Is the binary named tailscaled?

No, the binary is `/opt/myorg/bin/my-tailscale-program`, so `my-tailscale-program`.

I suspect the `dir` part of the logpolicy's cfgPath assignment is incorrect.
